### ISSUE TITLE
Read / write operation failed in HomeKit scripts after several minutes

### ISSUE THREAD

**Issue by seanwalter** [maxileith/homebridge-airport-express-connected#85]
### Analysis

When checking occupancy for these sensors in HomeKit scripts and iOS shortcuts, I get a Read/Write operation failed error after the child bridge has been running for a few to several minutes. Upon restarting the child bridge, everything works again for a few minutes.

I see this error in the plugin logs; not sure if it’s relevant, but it seems worth noting:

`(node:14447) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 packets listeners added to [EventEmitter].`

I have 14 Airport Expresses plus two Homebridge Minis, in case the plugin assumes a smaller number of devices.

Thank you for any advice/suggestions on resolving this.

### Expected Behavior

HomeKit scripts and iOS shortcuts should get occupancy value from plugin accessory without an error being generated.

### Steps To Reproduce

Install plugin. Allow it to run for a few to several minutes. Check occupancy status of one of the plugin accessories via HomeKit automation script or iOS shortcut.

### Logs

```Markdown
(node:14447) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 packets listeners added to [EventEmitter].
```

### Configuration

```JSON
{
            "name": "Airport Express Connected",
            "update": {
                "refreshRate": 3,
                "ignoreGroupWithLeadingDevice": true,
                "unreachable": {
                    "ignore": false,
                    "threshold": 30,
                    "reportDisconnected": true
                }
            },
            "discovery": {
                "enabled": true,
                "always": false,
                "intervals": 30,
                "whitelist": {
                    "list": [],
                    "enabled": false
                },
                "blacklist": {
                    "list": [],
                    "enabled": false
                },
                "discardKnownDevices": false
            },
            "_bridge": {
                "username": "xx:xx:xx:xx:xx:xx",
                "port": 44803
            },
            "platform": "AirportExpressConnected"
        },
```

### Environment

- OS: unraid 6.12.8 server
- Docker: 24.0.9
- Homebridge: 1.7.0 UI 4.55.1
- Airport Express Connected: 1.3.0 (have also tried latest beta and 2 earlier versions)
- Node: 20.11.1
- npm: 10.4.0

**Comment by maxileith** [comment]
Yeah, this is an issue with the current architecture that occurs when using the plugin with a lot of AirPort Expresses.

I am planning to fix that issue eventually. I will update you here. Currently, I cannot give you a timeframe.

**Comment by seanwalter** [comment]
Thank you. I’ll sit tight for now.

**Comment by seanwalter** [comment]
I came back to this recently and unchecked Enabled in the Discovery section of the configuration. Since doing that I haven’t gotten this error. So I’d consider that to be a reasonable fix for larger installations…scan once, discover devices, then turn off automatic discovery.
